ddtrace/internal/opentelemetry/context.py

```diff
@@ -69,21 +61,21 @@ def get_current(self):
         from opentelemetry.trace import NonRecordingSpan as OtelNonRecordingSpan
         from opentelemetry.trace import SpanContext as OtelSpanContext
         from opentelemetry.trace import set_span_in_context
-        from opentelemetry.trace.span import TraceFlags
         from opentelemetry.trace.span import TraceState
 
         from .span import Span
+        from .span import _get_trace_flags
 
         ddactive = self._ddcontext_provider.active()
         context = OtelContext()
         if isinstance(ddactive, DDSpan):
             span = Span(ddactive)
             context = set_span_in_context(span, context)
         elif isinstance(ddactive, DDContext):
-            tf = TraceFlags.SAMPLED if ddactive._traceflags == "01" else TraceFlags.DEFAULT
             ts = TraceState.from_header([ddactive._tracestate])
-            span_context = OtelSpanContext(ddactive.trace_id or 0, ddactive.span_id or 0, True, tf, ts)
-            span = OtelNonRecordingSpan(span_context)
+            tf = _get_trace_flags(ddactive.sampling_priority)
```

Comment:
I added these changes to resolve an edge case where a default sampling decision was set on traces (default is unsampled in otel). 

If these changes look good to you we can merge this PR